components/director/internal/domain/fetchrequest/service.go

```
@@ -0,0 +1,96 @@
+	return &service{
+		client:       client,
+		logger:       logger,
+		timestampGen: timestamp.DefaultGenerator(),
+	}
+}
+
+func (s *service) FetchAPISpec(fr *model.FetchRequest) (*string, *model.FetchRequestStatus) {
+
+	valid, msg := s.validateFetchRequest(fr)
+	if !valid {
+		return nil, s.fixStatus(model.FetchRequestStatusConditionInitial, msg)
+	}
+
+	resp, err := s.client.Get(fr.URL)
+	if err != nil {
+		s.logger.Errorf("While fetching API Spec: %s", err.Error())
+		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, fmt.Sprintf("While fetching API Spec: %s", err.Error()))
+	}
+
+	defer func() {
+		if resp.Body != nil {
+			err := resp.Body.Close()
+			if err != nil {
+				s.logger.Errorf("While closing body: %s", err.Error())
+			}
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		s.logger.Errorf("While fetching API Spec status code: %d", resp.StatusCode)
+		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode))
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		s.logger.Errorf("While reading API Spec: %s", err.Error())
+		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, fmt.Sprintf("While reading API Spec: %s", err.Error()))
+	}
+
+	spec := string(body)
+	return &spec, s.fixStatus(model.FetchRequestStatusConditionSucceeded, "")
+}
+
+func (s *service) validateFetchRequest(fr *model.FetchRequest) (bool, string) {
+	if fr.Mode != model.FetchModeSingle {
+		s.logger.Errorf("Unsupported fetch mode: %s", fr.Mode)
+		return false, fmt.Sprintf("Unsupported fetch mode: %s", fr.Mode)
+	}
+
+	if fr.Auth != nil {
+		s.logger.Error("Fetch Request Auth was provided")
+		return false, "Fetch Request Auth was provided"
+	}
+
+	if fr.Filter != nil {
+		s.logger.Error("Fetch Request Filter was provided")
+		return false, "Fetch Request Filter was provided"
+	}
+	return true, ""
```

- Consider returning just error (not bool & string)
- All error messages should clarify that these features are not supported.

- Maybe we should fail the FR when unsupported params are specified?